sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -740,6 +744,28 @@ case class Cast(
       } else {
         buildCast[UTF8String](_, s => DateTimeUtils.stringToTime(s).orNull)
       }
+
+    case TimestampType =>
```

Comment:
The scope of this PR should be adding the `time()` function alias for `CAST`. Let's implement CAST and test it separately, please.